lib/portage/package/ebuild/fetch.py

```diff
@@ -434,8 +564,9 @@ def fetch(myuris, mysettings, listonly=0, fetchonly=0,
 	for myfile, myuri in file_uri_tuples:
 		if myfile not in filedict:
 			filedict[myfile]=[]
-			for y in range(0,len(locations)):
-				filedict[myfile].append(locations[y]+"/distfiles/"+myfile)
+			for l in locations:
+				filedict[myfile].append(get_mirror_url(l, myfile,
+						mysettings["EROOT"]))
```

Comment:
Let's make this lazy by appending `functools.partial(get_mirror_url, l, myfile, mysettings["EROOT"])`, and later evaluate it like this:
```diff
--- a/lib/portage/package/ebuild/fetch.py
+++ b/lib/portage/package/ebuild/fetch.py
@@ -895,6 +895,8 @@ def fetch(myuris, mysettings, listonly=0, fetchonly=0,
                        tried_locations = set()
                        while uri_list:
                                loc = uri_list.pop()
+                               if isinstance(loc, functools.partial):
+                                       loc = loc()
                                # Eliminate duplicates here in case we've switched to
                                # "primaryuri" mode on the fly due to a checksum failure.
                                if loc in tried_locations:

```

Reply:
Done and pushed.